### PR TITLE
Removal of metrics code

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following removes all of the metrics code. This includes the SLA/Metric found on applications, models, and charms. This doesn't compile because we need to bring the charm repo into this repository as charms will no longer be metered.

This change is because we're not collecting metrics via romulus or omnibus. This code would have had to be migrated to dqlite. Removing this code just removes the necessity to perform that work.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

```sh
$ juju bootstrap lxd test --build-agent
$ juju add-model default
$ juju deploy ubuntu -n 3
```

## Documentation changes

This removes metrics code from juju/juju.

## Links

**Jira card:** JUJU-5497

### CODE REVIEW

**Comment by hmlanigan** [comment]
Can we split this into 2 PRs? one that adds charm if necessary? It feels like we're sneaking it in here.

File: state/migration_export.go [hunk 1]
```diff
@@ -238,9 +238,6 @@ func (st *State) exportImpl(cfg ExportConfig, leaders map[string]string, store o
 		}
 	}
 
-	export.model.SetSLA(dbModel.SLALevel(), dbModel.SLAOwner(), string(dbModel.SLACredential()))
-	export.model.SetMeterStatus(dbModel.MeterStatus().Code.String(), dbModel.MeterStatus().Info)
```
Comment (hmlanigan) [inline state/migration_export.go:242]: Have you updated the description package too?

**Review by hmlanigan — APPROVED** [review]
Found more code to be deleted I think:

internal/sender - the doc file indicates it's there for cmd/juju/metricsdebug and worker/metrics/sender, which you've removed.

A few comments of follow up. Otherwise LGTM.

File: state/migration_export.go (hunk 1 above)
Comment (SimonRichardson) [reply state/migration_export.go:242]: I haven't I'm not sure how we should tackle that one. At the very least it should be ok to just leave it, as we'll drop the metrics on the floor.

File: go.mod [hunk 2]
```diff
@@ -72,7 +72,6 @@ require (
 	github.com/juju/ratelimit v1.0.2
 	github.com/juju/replicaset/v3 v3.0.1
 	github.com/juju/retry v1.0.0
-	github.com/juju/romulus v1.0.0
```
[the remark this answers is not in the document]
Comment (SimonRichardson) [reply go.mod:75]: We should.

**Comment by SimonRichardson** [comment]
> internal/sender - the doc file indicates it's there for cmd/juju/metricsdebug and worker/metrics/sender, which you've removed.

It's actually the prometheus metrics (naming is hard).

**Pull request merged**